hydromt/models/_v1/model_root.py

```diff
@@ -0,0 +1,187 @@
+        for i, handeler in enumerate(self.logger.handlers):
+            # make the type checkers a little happier
+            handeler = cast(FileHandler, handeler)
+            log_level = handeler.level
+            if hasattr(handeler, "baseFilename"):
+                if dirname(handeler.baseFilename) != self._path:
+                    # remove handler and close file
+                    self.logger.handlers.pop(i).close()
+                else:
+                    has_log_file = True
+                break
+        if not has_log_file:
+            new_path = join(self._path, "hydromt.log")
+            add_filehandler(self.logger, new_path, log_level)
+
+    def _set_path(self, path: PathLike) -> None:
+        """Set the path of the root, adjusting for cross platform where necessary."""
+        if self._describes_windows_path(path):
+            pwp = PureWindowsPath(path)
+            if system() in ["Linux", "Darwin"]:
+                if not pwp.is_absolute():
+                    # posix path on a windows system
+                    abs_path = abspath(join(getcwd(), pwp.as_posix()))
+                else:
+                    drive = pwp.drive
+                    relative_path_part = PureWindowsPath(str(pwp).removeprefix(drive))
+                    drive = drive.replace(":", "").lower()
+                    abs_path = abspath(
+                        join("/mnt", drive, *(relative_path_part.parts[1:]))
+                    )
+                self._path: Path = Path(abs_path)
+            else:
+                # windows path on a windows system
+                self._path: Path = Path(abspath(pwp))
+
+        elif self._describes_posix_path(path):
+            ppp = PurePosixPath(path)
+            if system() == "Windows":
+                # posix path on a windows system
+                parts = ppp.parts
+                if self._posix_is_mounted(ppp):
+                    # were mounted somewhere so map the second dir to a drive
+                    drive = parts[1]
+                    relative_path_part = parts[2:]
+                    abs_path = abspath(join(f"{drive.upper()}:", *relative_path_part))
+                else:
+                    abs_ppp = Path(abspath(ppp))
+                    # abspath will almost certainly look like /home/user/whatever
+                    relative_path_part = abs_ppp.parts[3:]
+                    # in windows user dirs are usually titleized
+                    abs_path = abspath(
+                        join(f"C:\\Users\\{getlogin().title()}", *relative_path_part)
+                    )
+
+                self._path: Path = Path(abs_path)
+
+            else:
+                # posix path on a posix system
+                self._path: Path = Path(abspath(ppp))
+
+    def set_mode(self, mode: ModeLike) -> None:
+        """Set the mode of the model."""
+        self.mode: ModelMode = ModelMode.from_str_or_mode(mode)
+
+    def is_writing_mode(self) -> bool:
```

Comment:
I would add r+ aslo in the writing modes especially if you do not have here an override accessor.